src/Provider/Instagram.php

```diff
@@ -45,43 +45,88 @@ class Instagram extends OAuth2
     /**
      * {@inheritdoc}
      */
-    protected function initialize()
+    public function getUserProfile()
     {
-        parent::initialize();
+        $parameters = [
+            'fields' => 'id,username,account_type,media_count',
+            $this->accessTokenName => $this->getStoredData($this->accessTokenName),
+        ];
+
+        $response = $this->apiRequest('me', 'GET', $parameters);
+
+        $data = new Collection($response);
+
+        if (!$data->exists('id')) {
+            throw new UnexpectedApiResponseException('Provider API returned an unexpected response.');
+        }
 
-        // The Instagram API requires an access_token from authenticated users
-        // for each endpoint, see https://www.instagram.com/developer/endpoints.
-        $accessToken = $this->getStoredData($this->accessTokenName);
-        $this->apiRequestParameters[$this->accessTokenName] = $accessToken;
+        $userProfile = new User\Profile();
+
+        $userProfile->identifier = $data->get('id');
+        $userProfile->displayName = $data->get('username');
+        $userProfile->profileURL = "https://instagram.com/{$data->get('username')}";
+        $userProfile->data = [
+            'account_type' => $data->get('account_type'),
+            'media_count' => $data->get('media_count'),
+        ];
+
+        return $userProfile;
     }
 
     /**
-     * {@inheritdoc}
+     * Fetch user medias.
+     *
+     * @param int $limit Number of elements per page.
+     * @param string $pageId Current pager ID.
+     * @param array|null $fields Fields to fetch per media.
+     *
+     * @throws \Hybridauth\Exception\UnexpectedApiResponseException If API's response is not valid.
+     *
+     * @return \Hybridauth\Data\Collection Response encapsulated in a `Collection`.
      */
-    public function getUserProfile()
+    public function getUserMedia(int $limit = 12, ?string $pageId = null, array $fields = null)
```

Comment:
I like typehint here - but it won't work in php 5.4. so, remove int/string. and keep array only

Reply:
Fixed